### PR TITLE
F-/ISAPT-D

### PR DESCRIPTION
**Pull request by dsirianni**
## Description
Adds empirical dispersion capabilities to F-/ISAPT, while maintaining functional group partitioning capabilities

## Todos
Notable points (developer or user-interest) that this PR has or will accomplish.
- [ ] Add hooks in driver/QCEngine
- [ ] Add tests

## Checklist
- [ ] Tests added for any new features
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

### CODE REVIEW

File: psi4/share/psi4/fsapt/fsapt.py [hunk 1]
```diff
@@ -328,40 +392,150 @@ def printFrag(geom, Z, Q, fragkeys, frags, nuclear_ws, orbital_ws, filename):
...
+    #vals['Total'] = [[0.0 for x in vals['Elst'][0]] for x2 in vals['Elst']]
+    #for key in ['Elst', 'Exch', 'IndAB', 'IndBA', 'Disp']:
+    #    for k in range(len(vals['Total'])):
+    #        for l in range(len(vals['Total'][0])):
+    #            vals['Total'][k][l] += vals[key][k][l]
+
     return vals
 
-def extractOrder2Fsapt(osapt, wsA, wsB):
+def fragmentD3Disp(d3disp: np.ndarray, frags: Dict[str, Dict[str, List[str]]]) -> Tuple[float, Dict[str, Dict[str, float]]]:
```
Comment (dgasmith) [inline psi4/share/psi4/fsapt/fsapt.py]: It would be a good time to ensure pep8 snake_case function names.

Comment (dgasmith) [reply psi4/share/psi4/fsapt/fsapt.py]: @loriab Do we have 3.6 Python base to require typings?

Comment (loriab) [reply psi4/share/psi4/fsapt/fsapt.py]: yes, psi is 3.6+

**Comment by loriab** [comment]
continues on in #2081

**Pull request closed without merging**